### DISCUSSION TITLE
Integrated / bundled knex version

### DISCUSSION THREAD

**Discussion by er-imageline** [mikro-orm/mikro-orm#5230] [Q&A]
Hi,

I'm trying to hook the Elastic APM observability tool in to MikroOrm.

They list knex >=0.20.0 <3 as being compatible with [auto-instrumentation of ESM modules](https://www.elastic.co/guide/en/apm/agent/nodejs/current/esm.html#esm-compat-modules).

However the releases / versions between projects seem to be wildly different, so it seems these projects have diverged.  Is mikro-orm/knex now effectively a fork of https://github.com/knex/knex ?

**Comment by B4nan** [comment]
MikroORM v6 uses knex v3, so if you really want to use that tool, you can only use older versions. I dont understand what wild differences you see, there is some monkey-patching to fix some issues with knex (that the maintainers didnt really want to fix in the repo unfortunately), but that does not make it a fork.

**Reply by er-imageline** [reply]
Thanks for the quick reply!  And no offence intended by that remark - I just saw that the dates + release notes of the last few releases didn't seem to line up.

Glad to hear it uses v3.  Out of curiosity, is it possible in MikroOrm to specify the version of knex that it uses (either the integrated one or the separate project)?

**Reply by B4nan** [reply]
> I just saw that the dates + release notes of the last few releases didn't seem to line up.

That package is not a fork, its a knex specific implementation of a driver for the ORM, a common base for all knex-based drivers.

> Out of curiosity, is it possible in MikroOrm to specify the version of knex that it uses (either the integrated one or the separate project)?

Only via NPM overrides or alternatives (e.g. yarn resolutions), the version is pinned, mainly because I want to be in control of this - as the ORM is patching the code, and even a patch release could be breaking for us. Not to mention that some dependencies tend to do breaking stuff in patch releases regardless of this... Been there, done that.
